cmd/snap-confine/snap-confine.c

```diff
@@ -81,6 +81,9 @@ int main(int argc, char **argv)
 	if (!verify_security_tag(security_tag))
 		die("security tag %s not allowed", security_tag);
 
+	const char *snap_name = getenv("SNAP_NAME");
+	sc_snap_name_validate(snap_name, NULL);
```

Comment:
Per long discussion on the channel, I agree with @mvo5 here. This is magic, and the error system in snap-confine seems too complex for what we need. I suggest a simplification so that errors become simply:
```
error *sc_validate_snap_name(snap_name) {
        ...
        return errorf("something wrong happened");
}
```